recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,175 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+
+class SociConan(ConanFile):
+    name = "soci"
+    homepage = "https://github.com/SOCI/soci"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "The C++ Database Access Library "
+    topics = ("cpp", "database-library", "oracle", "postgresql", "mysql", "odbc", "db2", "firebird", "sqlite3", "boost" )
+    license = "BSL-1.0"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    _cmake = None
+
+    options = {
+        "fPIC":             [True, False],
+        "shared":           [True, False],
+        "empty":            [True, False],
+        "with_sqlite3":     [True, False],
+        "with_db2":         [True, False],
+        "with_odbc":        [True, False],
+        "with_oracle":      [True, False],
+        "with_firebird":    [True, False],
+        "with_mysql":       [True, False],
+        "with_postgresql":  [True, False],
+        "with_boost":       [True, False]
+    }
+
+    default_options = {
+        "fPIC":             True,
+        "shared":           False,
+        "empty":            False,
+        "with_sqlite3":     False,
+        "with_db2":         False,
+        "with_odbc":        False,
+        "with_oracle":      False,
+        "with_firebird":    False,
+        "with_mysql":       False,
+        "with_postgresql":  False,
+        "with_boost":       False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _validate_compiler_settings(self):
+        tools.check_min_cppstd(self, "11")
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5.0":
+            raise ConanInvalidConfiguration("gcc minimum required version is 5.0")
+        elif self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < "3.8":
+            raise ConanInvalidConfiguration("clang minimum required version is 3.8")
+        elif self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) < "10.0":
+            raise ConanInvalidConfiguration("apple-clang minimum required version is 10.0")
+        elif self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < "19":
+            raise ConanInvalidConfiguration("Visual Studio minimum required version is 19")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        self._validate_compiler_settings()
+
+    def requirements(self):
```

Comment:
why is there no boost requirement? https://github.com/SOCI/soci/blob/b208b22456698581996162688da27d300fa6651f/cmake/dependencies/Boost.cmake#L7